support-frontend/assets/helpers/subscriptionsForms/formReducer.js

```diff
@@ -79,13 +79,13 @@ function createFormReducer(
         return { ...state, titleGiftRecipient: action.titleGiftRecipient };
 
       case 'SET_FIRST_NAME_GIFT':
-        return { ...state, firstNameGiftRecipient: action.firstNameGiftRecipient, formErrors: removeError('firstName', state.formErrors) };
+        return { ...state, firstNameGiftRecipient: action.firstNameGiftRecipient, formErrors: removeError('firstNameGiftRecipient', state.formErrors) };
 
       case 'SET_LAST_NAME_GIFT':
-        return { ...state, lastNameGiftRecipient: action.lastNameGiftRecipient, formErrors: removeError('lastName', state.formErrors) };
+        return { ...state, lastNameGiftRecipient: action.lastNameGiftRecipient, formErrors: removeError('lastNameGiftRecipient', state.formErrors) };
 
       case 'SET_EMAIL_GIFT':
-        return { ...state, emailGiftRecipient: action.emailGiftRecipient };
+        return { ...state, emailGiftRecipient: action.emailGiftRecipient, formErrors: removeError('emailGiftRecipient', state.formErrors) };
```

Comment:
This is optional in the existing checkout, was a specific decision made to make it mandatory because optional feels more appropriate to me?

Reply:
I'm kind of relieved that you said this - it made me vaguely uncomfortable for this to be mandatory. I checked with @jesseyuen and he agrees it should be optional.